### PR TITLE
29dev

### PR DESCRIPTION
**Pull request by yutsevich**

### CODE REVIEW

**Review by julia-shvabovich — APPROVED** [review]
well done

File: src/main/java/com/internet/shop/controllers/InjectDataController.java [hunk 1]
```diff
@@ -25,9 +27,11 @@
...
             throws ServletException, IOException {
         User user1 = userService.create(new User("User1", "log1", "111"));
-        User user2 = userService.create(new User("User2", "log2", "222"));
-        ShoppingCart shoppingCart1 = shoppingCartService.create(new ShoppingCart(user1.getId()));
-        ShoppingCart shoppingCart2 = shoppingCartService.create(new ShoppingCart(user2.getId()));
+        user1.setRoles(Set.of(Role.of("USER")));
+        User admin = userService.create(new User("admin", "admin", "222"));
+        admin.setRoles(Set.of(Role.of("ADMIN")));
+        shoppingCartService.create(new ShoppingCart(user1.getId()));
+        shoppingCartService.create(new ShoppingCart(admin.getId()));
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/InjectDataController.java]: admin should not have his own shopping cart

also, let`s first set roles, and then save users to storage

File: src/main/java/com/internet/shop/controllers/RegistrationController.java [hunk 2]
```diff
@@ -33,6 +35,7 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
 
         if (password.equals(repeatedPassword)) {
             User user = userService.create(new User(name, login, password));
+            user.setRoles(Set.of(Role.of("USER")));
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/RegistrationController.java:38]: same here

File: src/main/java/com/internet/shop/model/Role.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package com.internet.shop.model;
+
+public class Role {
+    private Long id;
+    private RoleName roleName;
+
+    public Role(RoleName roleName) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/model/Role.java]:
```suggestion
    private Role(RoleName roleName) {
```

File: src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        protectedUrls.put("/admin/products", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/delete", List.of(Role.RoleName.ADMIN));
+
+        protectedUrls.put("/products/all", List.of(Role.RoleName.USER));
+        protectedUrls.put("/order", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-carts/products/add", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-carts/products/delete", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-carts", List.of(Role.RoleName.USER));
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java:41]: USER can see `/users/orders/all` as well

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/controllers/RegistrationController.java (hunk 2 above)
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/RegistrationController.java]: let`s first set roles, and then save user to storage

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**